### PR TITLE
Migrate webapp global state to redux

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

**Comment by harshilsharma63** [comment]
Some bugs I've found-

#### Cannot open board from shared URL.
1. Generate a share URL
2. Open the share URL in a different browser or in incognito.
3. Notice the error and the fact that the board doesn't open. See screenshot below-

Screenshot 2021-07-30 at 10 51 13 AM

#### Flashing error message when creating a new card
1. Open a board (any view)
2. Create a new card, either by the blue New button on the top right or some other way.
3. Notice in the card popup that opens, an error message flashes for a split second. See view below-

https://user-images.githubusercontent.com/18575143/127604735-d6a1f449-96d7-4558-b7ea-70896e789f64.mp4

#### Flashing WebSocket connection on page reloading
1. Open a board
2. Reload the page.
3. Notice the WebSocket connection error flashes on the top. See video below

https://user-images.githubusercontent.com/18575143/127605192-064a2683-775b-4fed-8d10-5fa1cff9d8c4.mp4

**Comment by sbishel** [comment]
I tested this locally and appears that all bugs Harshil found have been fixed. Everything else worked good. I didn't find any other issues.

File: webapp/src/components/centerPanel.tsx [hunk 1]
```diff
@@ -94,16 +102,13 @@ class CenterPanel extends React.Component<Props, State> {
...
-        if (!groupByProperty && activeView.viewType === 'board') {
-            Utils.assertFailure('Board views must have groupByProperty set')
-            return <div/>
-        }
-
-        const {board} = boardTree
+        // if (!groupByProperty && activeView.fields.viewType === 'board') {
+        //     Utils.assertFailure('Board views must have groupByProperty set')
+        //     return <div/>
+        // }
```
Comment (sbishel) [inline webapp/src/components/centerPanel.tsx:111]: Remove if not necessary.

**Review by sbishel — APPROVED** [review]
Approving for merge, but need to add unit tests.

**Pull request merged**